### PR TITLE
concrete implementation for Learning to Rank objects

### PR DESCRIPTION
**Pull request by gcmgomes**
Will follow up with the actual implementation in a future PR, just want to know what you guys think. Could you please take a look at it @raphaelcampos ?

### CODE REVIEW

File: base/points/BUILD [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package(default_visibility = ["//visibility:public"])
+
+cc_library(
+  name =  "l2r_point",
+  hdrs = ["l2r_point.h"],
+  srcs = ["l2r_point.cc"],
```
Comment (raphaelcampos) [inline base/points/BUILD]: Didn't we define the file extensions as .h and .cpp?

File: base/points/l2r_point.h [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+ * suitable concrete class to handle it.
+ */
+class L2RPoint : public ::base::Point {
+ public:
+  /**
+   * \brief Standard destructor.
+   */
+  ~L2RPoint();
+
+  /** Returns the feature indexed by the received parameter. Indexing is 1
```
Comment (raphaelcampos) [inline base/points/l2r_point.h]: "Returns the feature value..." would be better.

**Comment by raphaelcampos** [comment]
the .cc it is just a mock implementation, right? 
I didn't miss anything in the idea. Perhaps, it would be better someone that works with L2R give a look at it as well. @luizfgoncalves or @clebsonc could you guys check it out? Give your opinions...

**Comment by luizfgoncalves** [comment]
In general it looks good to me, I don' think there is anything important missing.
Just two things:
1º You used string for query and double for label. Shouldn't both be double? As queries usually are represented by numbers too. Or, maybe, both could be strings too (You can find some datasets that use words or letters for it). What do you guys think is better?

2º You created only a vector to store the feature falues, so you are assuming the dataset will be dense. I get that most L2R datasets are this way, but maybe a more generic approach (that supports dense and sparse data) could be better.

**Comment by gcmgomes** [comment]
We usually need labels to be numeric, specially since we use MSE heavily during our computations, and casting back and forth from string to double would take way too much of our processing time. As for the query, TD2004 uses string identifiers for these, hence the string value.

We could just implement a `l2r_sparse_point?` later on. Using hashes, despite being asymptotically  fast, we would still need to compute hash functions whenever retrieving a feature value. Let's avoid those overheads if we can, and get ourselves a nice speedup.

File: base/points/BUILD (hunk 1 above)
Comment (gcmgomes) [reply base/points/BUILD]: Yeah, my bad. Fixed.

**Comment by raphaelcampos** [comment]
LGTM

**Comment by gcmgomes** [comment]
Thanks!

**Pull request merged**